src/app/shared/classes/base-page.ts

```diff
@@ -0,0 +1,48 @@
+import { Platform } from '@ionic/angular';
+import { AuthenticationProvider } from '../../providers/authentication/authentication';
+import { NavigationExtras, Router } from '@angular/router';
+
+export abstract class BasePageComponent {
+
+  protected constructor(
+    public platform: Platform,
+    public authenticationProvider: AuthenticationProvider,
+    public router: Router,
+    public loginRequired: boolean = true,
+  ) {
+
+  }
+
+  ionViewWillEnter() {
```

Comment:
shouldn't this sort of thing be covered in the route guard now? that's essentially what this is